mule-deployer/src/test/java/org/mule/tools/client/standalone/controller/UnixControllerTest.java

```diff
@@ -0,0 +1,113 @@
+import static org.powermock.api.mockito.PowerMockito.doReturn;
+import static org.powermock.api.mockito.PowerMockito.spy;
+
+@RunWith(PowerMockRunner.class)
+@PrepareForTest({UnixController.class})
+public class UnixControllerTest {
+
+  private static final String MULE_HOME = "/home/mule";
+  private static final int TIMEOUT = 3000;
+  private final UnixController controller = new UnixController(MULE_HOME, TIMEOUT);
+  private UnixController controllerSpy;
+  private static String CONTROLLER_ARG = "argValue";
+  private static final String[] CONTROLLER_ARGUMENTS = {CONTROLLER_ARG};
+  private static final int RUNNING_STATUS = 0;
+  private static final int NOT_RUNNING_STATUS = 1;
+
+  @Rule
+  public ExpectedException expected = ExpectedException.none();
+
+  @Before
+  public void setUp() throws Exception {
+    controllerSpy = spy(controller);
+  }
+
+  @Test
+  public void getMuleBinTest() {
+    assertThat("Mule bin path is not the expected", controller.getMuleBin(), equalTo("/home/mule/bin/mule"));
+  }
+
+  @Test
+  public void statusRunningTest() throws Exception {
+    doReturn(RUNNING_STATUS).when(controllerSpy, "runSync", "status", CONTROLLER_ARGUMENTS);
+    assertThat("Status is not the expected", controllerSpy.status(CONTROLLER_ARGUMENTS), equalTo(RUNNING_STATUS));
+  }
+
+  @Test
+  public void statusNotRunningTest() throws Exception {
+    doReturn(NOT_RUNNING_STATUS).when(controllerSpy, "runSync", "status", CONTROLLER_ARGUMENTS);
+    assertThat("Status is not the expected", controllerSpy.status(CONTROLLER_ARGUMENTS), equalTo(NOT_RUNNING_STATUS));
+  }
+
+  @Test
+  public void getProcessIdIfMuleIsRunningTest() {
+    int processId = 1;
+    doReturn(true).when(controllerSpy).isMuleRunning();
+    doReturn(processId).when(controllerSpy).getProcessIdFromStatus();
+    assertThat("Process id is not the expected", controllerSpy.getProcessId(), equalTo(processId));
+  }
+
+  @Test
+  public void getProcessIdMuleProcessNotRunningExceptionTest() {
+    expected.expect(MuleControllerException.class);
+    expected.expectMessage("Mule Runtime is not running");
+    doReturn(false).when(controllerSpy).isMuleRunning();
+    controllerSpy.getProcessId();
+  }
+
+  @Test
+  public void getProcessIdFromMuleStatusTest() throws Exception {
+    int processId = 10;
+    setStatusToOutputStreamInController("Mule is running (" + processId + ").");
+    assertThat("Process id is not the expected", controllerSpy.getProcessIdFromStatus(), equalTo(processId));
+  }
+
+  @Test
+  public void getProcessIdFromMuleEEStatusTest() throws Exception {
+    int processId = 10;
+    setStatusToOutputStreamInController("Mule Enterprise Edition is running (" + processId + ").");
+    assertThat("Process id is not the expected", controllerSpy.getProcessIdFromStatus(), equalTo(processId));
+  }
+
+  @Test
+  public void getProcessIdFromStatusExceptionTest() throws Exception {
+    expected.expect(MuleControllerException.class);
+    expected
```

Comment:
Please stop asserting messages of exceptions unless really necessary this method throws MuleControllerException for two reasons, it should be enough to make sure in the test that you're validating one or the other. If you think that's not clear enough may be we should thrown an exception wrapped inside the MuleControllerException